### PR TITLE
feat(ilp): make ILP implicit schema optional

### PR DESCRIPTION
**Pull request by ideoma**
Adds 2 configuration options:

```
# Enable / Disable  automatic creation of new columns in existing tables via ILP. When set to false overrides value of line.auto.create.new.tables to false
#line.auto.create.new.columns=true

# Enable / Disable automatic creation of new tables via ILP.
#line.auto.create.new.tables=true
```

Done for production hardening where new tables should not be created dynamically when a client sends bad formatted data to ILP socket.

As a side enhancement `TableReader` when created will check that table folder exists on first failure opening metadata. This is to fix retrying to open metadata file and timing out on deleted / non existing tables

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 57 / 64 (89.06%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|4|6|66.67%|
|:large_blue_circle:|io/questdb/cutlass/line/udp/LineUdpParserImpl.java|14|19|73.68%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableReader.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/udp/LineUdpLexer.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/udp/DefaultLineUdpReceiverConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultLineTcpReceiverConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|8|8|100.00%|

File: core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java [hunk 1]
```diff
@@ -264,6 +270,16 @@ private TableUpdateDetails getTableUpdateDetailsFromSharedArea(@NotNull NetworkI
             } else {
                 int status = engine.getStatus(securityContext, path, tableNameUtf16, 0, tableNameUtf16.length());
                 if (status != TableUtils.TABLE_EXISTS) {
+                    if (!autoCreateNewTables) {
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java:273]: can we please reuse the code? same code and error messages are in `initCacheRecord()`

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

**Comment by newskooler** [comment]
Would this bring performance delays for the data? Probably they would be insignificant, but still wondering.

**Comment by puzpuzpuz** [comment]
> Would this bring performance delays for the data? Probably they would be insignificant, but still wondering.

@newskooler there should be no noticeable performance impact. This commit adds a few branches that would always yield same execution flow since the config props don't change dynamically. Never taken branches are almost free on modern CPUs.

File: core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java (hunk 1 above)
Comment (ideoma) [reply core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java:273]: `initCacheEntry` is in Line UDP family. There is quite some duplications between line TCP and UDP. How would you suggest to reuse exception building code between them?

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 68 / 68 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableReader.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/udp/LineUdpLexer.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/udp/DefaultLineUdpReceiverConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/udp/LineUdpParserImpl.java|23|23|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultLineTcpReceiverConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|8|8|100.00%|

**Pull request merged**